dhall-docs/README.md

```diff
@@ -97,6 +60,281 @@ dhall-docs --input . --output-link ${OTHER_LINK}
+determined by the column of the `{` character from the opening brace. For
+example, on the following block comment:
+
+```dhall
+{-|
+foo
+  bar
+    baz
+-}
+```
+
+... and the following:
+
+```dhall
+              {-|
+              foo
+                bar
+                  baz
+                        -}
+```
+
+
+... will extract the contents of the comment, stripping the `{` leading whitespace
+from all other lines, returning the following text:
+
+```
+foo
+  bar
+    baz
+```
+
+A block comment that has a line with less indentation than the `{` character
+column one is not a valid `dhall-docs` comment. The following:
+
+```dhall
+    {-|
+    foo
+  bar
+        baz -}
+```
+
+although it's a valid Dhall block comment, it's not valid for `dhall-docs`. The
+same applies for block comments that starts on the same line as the opening
+brace. The following:
+
+```dhall
+{-| foo
+    bar
+    -}
+```
+
+is invalid.
+
+See that the closing brace (`-}`) can be in a new line or in the last one:
+`dhall-docs` will, in any case, insert a newline character (`\n`) in the last
+line.
+
+This is an example of an actual file documented using block comments:
+
+```dhall
+{-|
+# My awesome package
+
+This is the header of my package
+
+* foo
+  - bar
+  - baz
+
+qux
+
+    an indented code block
+-}
+let myAwesomeFunction {-|
+                      an example of a documentation -}
+               = \(x : Natural) -> x + 1
+
+in
+{
+  {-|
+  foo
+      bar
+          baz
+  -}
+  myAwesomeFunction
+}
+```
+
+If a block comment doesn't starts with `{-|` it will be ignored by `dhall-docs`.
+On the other hand, if a block comments starts with `{-|` and it doesn't follow
+the rules on the specification, it will be ignored _and_ a warning will be
+logged by `dhall-docs`.
+
+### Line comments
+
+Normal Dhall single-line comments starts with `--` and ends at a newline. For
+example:
+
+```dhall
+-- single line comment
+0
+```
+
+A `dhall-docs` valid single-line comment starts with `--|` and a single whitespace
+character. For example:
+
+```dhall
+--| foo
+```
+
+You can span your documentation in several line comments by writing `--` and two
+(2) whitespaces:
+
+```dhall
+--| foo
+--  bar
+--      baz
+```
+
+The `--` prefix defines the indentation for any line. That is, on the above
+example and the following
+
+```dhall
+--| foo
+  --  bar
+ --      baz
+```
+
+... `dhall-docs` will extract the following contents:
+
+```
+foo
+bar
+    baz
+```
+
+If you would like to force an empty line on the documentation output, place an
+empty single line comment in that place. On this example:
+
+```dhall
+--| foo
+--
+--  bar
+```
+
+`dhall-docs` will extract:
+
+```dhall
+foo
+
+bar
+```
+
+The empty `--` line doesn't need the two (2) whitespaces.
+
+If you left an empty line on the source code between two lines, it will be ignored.
+On this example:
+
+```dhall
+--| foo
+
+--  bar
+
+```
+
+`dhall-docs` will extract:
+
+```
+foo
+bar
+```
```

Comment:
I'd rather suggest treating `bar` as a separate non-documentation comment. Otherwise I think we'll find "internal" comments appended to the documentation.